src/main/java/pages/LoginPage.java

```diff
@@ -0,0 +1,110 @@
+package pages;
+
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.jsoup.helper.Validate.fail;
+
+public class LoginPage extends BasePage<LoginPage> {
+
+    @FindBy(xpath="//*[@data-test='username']")
+    WebElement usernameInput;
+    @FindBy(xpath="//*[@data-test='password']")
+    WebElement userpasswordInput;
+    @FindBy(id = "login-button")
+    WebElement loginBtn;
+    @FindBy(xpath = "//*[@id='login_credentials']")
+    WebElement listLogin;
+    @FindBy(xpath = "//*[@id='login_button_container']//*[@data-test='error']")
+    WebElement errorMessage;
+    @FindBy(xpath = "//*[@id=\"login_button_container\"]//button[@class='error-button']")
+    WebElement errorMessageCloseBtn;
+    @FindBy(xpath = "//*[@id='login_button_container']")
+    WebElement loginContent;
+
+    public LoginPage(WebDriver driver) {
+        super(driver);
+    }
+
+    @Override
+    LoginPage waitForPageOpened() {
+        return this;
+    }
+
+    /**
+     * Login to the site
+     *
+     * @param username
+     * @param password
+     */
+    public void login(String username, String password) {
+        usernameInput.sendKeys(username);
+        userpasswordInput.sendKeys(password);
+        loginBtn.click();
+    }
+
+    /**
+     * Open login page
+     */
+    public LoginPage openPage() {
+        load();
+        isLoaded();
+        return this;
+    }
+
+    public List<String> getListLogin(){
+        return Arrays.asList(listLogin.getText()
+                .replace("Accepted usernames are:\n", "").split("\n"));
```

Comment:
how to get all possible logins into an array and avoid "Accepted usernames are: \ n" and .split ("\ n")?

Reply:
in this case it's not easy to do that, you did it correctly using .replace. That is acceptable approach.